contracts/interchain-token-service/src/interface.rs

```diff
@@ -71,6 +115,25 @@ pub trait InterchainTokenServiceInterface: AxelarExecutableInterface {
         flow_limit: Option<i128>,
     ) -> Result<(), ContractError>;
 
+    /// Deploys an interchain token on the current chain.
+    ///
+    /// This function deploys a new interchain token with specified metadata and optional
+    /// initial supply. If initial supply is provided, it is minted to the caller. The
+    /// caller can also specify a separate minter address.
+    ///
+    /// # Arguments
+    /// - `caller`: Address of the caller initiating the deployment. The caller must authenticate.
+    /// - `salt`: A 32-byte unique salt used for token deployment.
+    /// - `token_metadata`: Metadata for the new token (name, symbol, decimals).
+    /// - `initial_supply`: Initial amount to mint to caller, if greater than 0.
+    /// - `minter`: Optional address that will have minting rights after deployment.
+    ///
+    /// # Returns
+    /// - `Result<BytesN<32>, ContractError>`: On success, returns the token ID (`BytesN<32>`).
+    /// - On failure, returns a `ContractError`.
```

Comment:
```suggestion
    /// - `Ok(BytesN<32>)`: Returns the token ID.
```
perhaps. contract error doc is trivial